Gems/ROS2/Code/Source/RobotImporter/Pages/PrefabMakerPage.cpp

```diff
@@ -60,5 +96,24 @@ namespace ROS2
     {
         return m_useArticulation->isChecked();
     }
+    AZStd::optional<AZ::Transform> PrefabMakerPage::getSelectedSpawnPoint() const
+    {
+        if (!m_spawnPointsInfos.empty())
+        {
+            int vectorIndex = m_spawnPointsComboBox->currentData().toInt();
+            AZStd::string mapKey = AZStd::string(m_spawnPointsComboBox->currentText().toStdString().c_str());
+            auto& map = m_spawnPointsInfos[vectorIndex];
+            auto spawnInfo = map.find(mapKey);
+            if (spawnInfo == m_spawnPointsInfos[vectorIndex].end())
+            {
+                return AZStd::nullopt;
+            }
+            return spawnInfo->second.pose;
+        }
+        else
+        {
+            return AZStd::nullopt;
+        }
+    }
```

Comment:
This section can be simplified
```suggestion
            if (auto spawnInfo = map.find(mapKey);
                spawnInfo != map..end())
            {
                return spawnInfo->second.pose;
            }
        }
        return AZStd::nullopt;
    }
```